packages/alfa-rules/src/sia-r83/rule.ts

```diff
@@ -0,0 +1,143 @@
+import { Rule } from "@siteimprove/alfa-act";
+import { Length } from "@siteimprove/alfa-css";
+import { Device } from "@siteimprove/alfa-device";
+import { Element, Text, Namespace, Node } from "@siteimprove/alfa-dom";
+import { Iterable } from "@siteimprove/alfa-iterable";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Ok, Err } from "@siteimprove/alfa-result";
+import { Style } from "@siteimprove/alfa-style";
+import { Page } from "@siteimprove/alfa-web";
+
+import { expectation } from "../common/expectation";
+
+import { hasAttribute } from "../common/predicate/has-attribute";
+import { isVisible } from "../common/predicate/is-visible";
+
+const { and, or, not, equals, test } = Predicate;
+const { isElement, hasNamespace } = Element;
+const { isText } = Text;
+
+export default Rule.Atomic.of<Page, Text>({
+  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r83.html",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return visit(document);
+
+        function* visit(node: Node, collect = false): Iterable<Text> {
+          if (
+            test(
+              and(
+                isElement,
+                or(
+                  hasAttribute("aria-hidden", equals("true")),
+                  not(hasNamespace(Namespace.HTML))
+                )
+              ),
+              node
+            )
+          ) {
+            return;
+          }
+
+          if (test(and(isText, isVisible(device)), node)) {
+            if (collect) {
```

Comment:
Wondering if `collect` should be moved up in the main test to benefit from lazy test evaluation and avoid the more costly `isText` and `isVisible`. Probably not a big gain…
```suggestion
          if (collect && test(and(isText, isVisible(device)), node)) {
```